<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

    <title>Staistical Phylogenetics</title>

    <link rel="stylesheet" href="../reveal.js/css/reveal.css">
    <link rel="stylesheet" href="../workshop_theme.css">
    <!--link rel="stylesheet" href="../../../talks/web_based/WinterBootcamp/wb_theme.css"-->

    <!-- Theme used for syntax highlighting of code -->
    <link rel="stylesheet" href="../reveal.js/lib/css/zenburn.css">

    <!-- Printing and PDF exports -->
    <script>
      var link = document.createElement( 'link' );
      link.rel = 'stylesheet';
      link.type = 'text/css';
      link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
      document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>
  </head>
  <body>
    <div class="reveal">
      <div class="slides">
        <section class="titlepage">
          <div class="meeting">
            mEpiLab BEAST Workshop, September 5th 2016
          </div>
               
          <div class="title">
            <h1>Statistical Phylogenetics</h1>
          </div>

          <div class="authors">
            Tim Vaughan
          </div>

          <div class="institution">
            Centre for Computational Evolution<br>
            The University of Auckland
          </div>

        </section>

        <section>
          <section class="center">
            <h2>Phylogenetic Trees</h2>
          </section>

          <section>
            <h3>What is a Phylogenetic Tree?</h3>

            <div class="fragment">
            <blockquote style="text-align:justify;margin-top:100px">
              A phylogenetic tree is a branching diagram showing
              the inferred evolutionary relationships among various
              biological species or other entities (their phylogeny)
              based on similarities and differences in their physical
              or genetic characteristics.
            </blockquote>
            <div class="cite">Wikipedia</div>
            </div>
          </section>

          <section>
            <h3>A short history</h3>

            <div style="text-align:center">
              <img style="width:38%" data-src="chain_of_being.png">
              <div class="cite">Rhetorica Christina, 1597</div>
            </div>
          </section>


          <section>
            <h3>A short history (continued)</h3>

            <div style="text-align:center">
              <img style="width:70%" data-src="Edward_Hitchcock_Paleontological_Chart.jpg">
              <div class="cite">Edward Hitchcock, Elementary Geology (1840)</div>
            </div>
          </section>

          <section>
            <h3>A short history (continued)</h3>

            <div class="figure inline">
              <img style="height:500px" data-src="Darwin_Tree_1837.png">
              <div class="cite">Darwin, Notebook B, 1837</div>
            </div>
            <div class="figure inline">
              <img style="height:500px" data-src="Darwin_Tree_OoS.png">
              <div class="cite">Origin of Species,1859</div>
            </div>

          </section>

          <section>
            <h3>Trees and Data</h3>

            <div class="figure" style="float:right;width:50%">
              <img data-src="primate_tree.jpg">
            </div>

            <div style="width:50%">
              <ul>
                <li class="fragment">Trees are inferred from characteristics of the
                entities they relate.</li>
                <li class="fragment">Characteristics are now often genetic, but may also be phenotypic.</li>
                <li class="fragment">The true tree constrains the distribution of these characteristics.</li>
              </ul>
            </div>
          </section>
          <section>
            <h3>Species and Gene Trees</h3>

            <div class="figure" style="float:right;width:50%">
              <img data-src="species_gene.jpg">
              <div class="cite">Knowles, Sys Biol, 2009</div>
            </div>

            <div style="width:50%">
              <ul>
                <li class="fragment">Many phylogenetic trees can be classed as
                either <b>species</b> or <b>gene</b> trees.</li>
                <li class="fragment">Gene trees are nested within species trees.</li>
                <li class="fragment">While species trees constrain gene trees, topologies may differ</li>
              </ul>
            </div>

            <p class="fragment" style="color:purple;text-align:center;margin-top:50px">Very similar
            relationship between transmission trees and pathogen
            trees.</p>
          </section>

          <section>
            <h3>The Neutrality Assumption</h3>

            In statisical phylogenetics, the common assumption is that
            the shape of the tree affects the distribution of
            characters, <b>but not the other way around</b>.

            <div class="figure fragment" style="text-align:center">
              <img style="width:75%" data-src="neutrality_tree.svg">
            </div>

            <p class="fragment">We are thus concerned only with <b>neutral</b> evolution.</p>
          </section>

          <section>
            <h3>The phylogenetic inference problem</h3>

            <blockquote>Given a set of $M$ sampled molecular
              sequences, infer the evolutionary
              relationships between them.</blockquote>

            <img class="fragment" data-src="primate_align.png"/>

            <p class="fragment">In a sense this is a just a clustering problem...</p>

          </section>

          <section>
            <h3>The phylogenetic inference problem</h3>

            <div style="text-align:center">
              <img data-src="primate_nj.svg" height="450"/>
            </div>

            <ul>
              <li class="fragment">Edge lengths may represent genetic distance  or time (if the molecular clock rate is known).</li>
              <li class="fragment">How can we do better than this?</li>
            </ul>
          </section>
        </section>

        <section>
          <section class="center">
            <h2>Modelling Molecular Evolution</h2>
          </section>

          <section>
            <h3>Genetic drift</h3>

            <blockquote>The random fluctuation in the frequency of
              a trait or allele (gene variant) distribution within a
              population due to the demographic process.</blockquote>


            <img class="fragment" style="vertical-align: middle; width: 400px; margin-right: 20px" data-src="drift1.svg"/>
            <img class="fragment" style="vertical-align: middle; width: 400px" data-src="drift2.svg"/>
          </section>

          <section>
            <h3>Modelling neutral genetic drift</h3>

            <!-- Wright-Fisher model -->
            <div style="position:relative; width:800px; height:400px; margin:0 auto;">
              <img class="fragment current-visible" width="800" data-src="WF5.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment current-visible" width="800" data-src="WF4.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment current-visible" width="800" data-src="WF3.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment current-visible" width="800" data-src="WF2.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment" width="800" data-src="WF.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
            </div>
            <ul>
              <li class="fragment">All individuals descend from a common ancestor at some point in the past.</li>
              <li class="fragment">Probability that the descendants of a given individual "take over" is $1/2N$.</li>
            </ul>

          </section>

          <section>
            <h3>The molecular clock hypothesis</h3>

            <br>

            <ol>
              <li class="fragment">In a single generation a new allele is produced in the population with probability $2N\mu$.</li>
              <li class="fragment">Any individual gene eventually becomes fixed in the population with probability $1/2N$.
            </ol>

            <br>
            <br>

            <blockquote class="fragment">
              The probability per generation that a new allele fixes
              is therefore $\mu$: independent of population size!
            </blockquote>

            <br>

            <blockquote class="alert fragment">This is not exactly true.</blockquote>
          </section>

          <section>
            <h3>Markov chains</h3>

            <p>Suppose we have a sequence of discrete variables $M_1,M_2,M_3\ldots$ where each variable
            can take values between 1 and $V$. The probability
            distribution over the first value is $P(M_1=m)\equiv p_1(m)$.</p>


            <p class="fragment">Subsequent probability distributions are related by:
            $$
            p_{i+1}(m')=\sum_{m=0}^{V}T_{mm'}p_{i}(m)
            $$</p>

            <p class="fragment">That is,
            $$
            p_2(1) = T_{11}p_1(1) + T_{21}p_1(2) + \ldots + T_{V1}p_1(V)
            $$</p>

            <p class="fragment" style="text-align:center;color:purple;margin-top:20px">This
            is a stochastic process called a <b>Markov chain</b> and
            $T_{mm'}$ is called the <b>transition matrix</b>.</p>
          </section>

          <section>
            <h3>Substitution Models</h3>

            Substitution is modeled as a (continuous time) Markov chain:

            <img class="fragment" style="width:100%;box-shadow:none" data-src="subst_model.svg">

            <blockquote class="fragment">Almost the only difference between different substitution
            models is the transition matrix.</blockquote>

            <p class="fragment">Usually the average rate of substitution ($\mu$) is factored out of the
            transition matrix.</p>
          </section>

          <section>
            <h3>The Jukes-Cantor Model</h3>

            The Jukes-Cantor model is the simplest (and least realistic) substitution model.

            <blockquote class="fragment" style="text-align:center;margin-top:50px">
              $T_{ij}$ is constant for all $i \neq $j.
            </blockquote>

            <ul style="margin-top:50px">
              <li class="fragment">Point mutations are equally likely between any pair of nucleotides.</li>
              <li class="fragment">No difference between transitions and transversions.</li>
              <li class="fragment">Equilibrium base frequencies are uniform.</li>
            </ul>

            <p class="fragment" style="margin-top:20px">Despite this, Jukes-Cantor is
            sometimes useful due to its simplicity.</p>
            
          </section>
          <section>
            <h3>Distances and Saturation</h3>

            Under a Jukes-Cantor model, the probability over character states as a function of time can be computed exactly:

            $$
            p_{diff}(t)=\frac{3}{4} - \frac{3}{4}e^{-\frac{4}{3}t}
            $$

            <div class="fragment" style="width:80%;text-align:center">
              <img style="width:80%" data-src="jc_probs.png">
            </div>

            <p class="fragment">Can be used to estimate distance: $\hat{t}=-\frac{3}{4}\log(1-\frac{4}{3}q)$.</p>
          </section>

          <section>
            <h3>The Tree Likelihood</h3>

            <blockquote>The tree likelihood $P(D|T,\sigma)$ is the probability that the observed sequences were produced by evolving a single ancestral sequence down $T$ under a neutral substitution model with parameters $\sigma$.</blockquote>

            <img class="fragment" data-src="likelihood.svg" width="800">

            <p class="fragment">There is an efficient algorithm for computing this
              (summing over all possible ancestral sequences):
              <b>Felsenstein's pruning algorithm</b>.</p>
          </section>


          <section>
            <h3>Maximum Likelihood Inference</h3>

            <blockquote>Maximum likelihood phylogenetic inference methods produce a
              single <b>point estimate</b> of the phylogenetic tree
              by seeking to maximize $P(D|T,\sigma)$ with respect
              to the tree $T$.</blockquote>
            
            <ul>
              <li class="fragment">Popular implementations of this method include PhyML and RAxML.</li>
              <li class="fragment">These methods are <i>fast</i>.</li>
              <li class="fragment">Point estimates provide no direct indication of the quality
              of the estimate - forced to used ad hoc methods such as
              bootstrap to remedy this.</li>
              <li class="fragment">ML does not allow the incorporation of available prior information.</li>
            </ul>
          </section>

          <section>
            <h3>The HKY Substitution Model</h3>

            <blockquote>
              Continuous time transition matrix:
            $$
            Q = \left(\begin{array}{cccc}
            * & \kappa\pi_C & \pi_A & \pi_G \\
            \kappa\pi_T & * & \pi_A & \pi_G \\
            \pi_T & \pi_C & * & \kappa\pi_G \\
            \pi_T & \pi_C & \kappa\pi_A & *
            \end{array}\right)
            $$
            </blockquote>
            <div class="cite">Hasegawa, Kishino, Yano (1985)</div>

            <ul style="margin-top:50px">
              <li class="fragment">Non-equlibrium base frequencies.</li>
              <li class="fragment">Includes transition/transversion rate ratio $\kappa$.</li>
              <li class="fragment">The simplest sensible model of neutral DNA evolution.</li>
            </ul>

          </section>
          <section>
            <h3>The GTR Substitution Model</h3>

            <blockquote>
              Continuous time transition matrix:
            $$
            Q = \left(\begin{array}{cccc}
            * & \alpha\pi_C & \beta\pi_A & \gamma\pi_G \\
            \alpha\pi_T & * & \delta\pi_A & \epsilon\pi_G \\
            \beta\pi_T & \delta\pi_C & * & \eta\pi_G \\
            \gamma\pi_T & \epsilon\pi_C & \eta\pi_A & *
            \end{array}\right)
            $$
            </blockquote>
            <div class="cite">Simon Tavar&eacute;, 1986</div>

            <ul style="margin-top:50px">
              <li class="fragment">The most general time-reversible model.</li>
              <li class="fragment">Ten parameters in total (9 if the average
              substiution rate is factored out).</li>
              <li class="fragment">All other time-reversible models are a special case of this.</li>

            </ul>

          </section>
          <section>
            <h3>Glaring Omissions</h3>

            <blockquote style="text-align:center;margin-top:50px">How great are these
            fancy models of substitution?</blockquote>

            <blockquote class="fragment alert" style="text-align:center;margin-top:50px">Not
            great!  There are many omissions!</blockquote>

            <ul style="margin-top:50px">
              <li class="fragment">Don't model insertions/deletions.</li>
              <li class="fragment">Don't model recombination.</li>
              <li class="fragment">No allowance for rate variation between sites or over tree.</li>
            </ul>

            <p class="fragment">There are several ways to address the last two points, some
            of which are implemented in BEAST.</p>

          </section>
        </section>

        <section>
          <section class="center">
            <h2>Modelling Genealogies</h2>
          </section>

          <section>
            <h3>The genealogical process</h3>

            <ul>
              <li>In the absence of selection, we can consider the
              process of tree generation in isolation.</li>

              <li class="fragment">If we are to take models of population genetics
              seriously, there is a full true species/gene tree and it
              is produced by a forward-in-time branching process.</li>

              <li class="fragment">Since we usually only ever sample a subset of the entities in
              this complete tree, the genealogy relating our samples
              is constructed backward-in-time from our
              samples.</li>

              
            <div style="position:relative; width:900px; height:150px; margin-top:20px;">
              <img class="fragment current-visible" width="900" data-src="full_tree.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0"/>
              <img class="fragment" width="900" data-src="sampled_tree.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0"/>
            </div>

          </section>

          <section>
            <h3>The tree prior</h3>

            <blockquote>The tree prior $P(T|\theta,l)$ is the
            probability (density) of a tree $T$ conditional on a
            genealogical process with parameters $\theta$ and
            additional sample data $l$ that is not part of the
            alignment.</blockquote>

            <ul>
              <li class="fragment">The tree prior is where much of the diversity in Bayesian phylogenetics resides.</li>
              <li class="fragment">Choosing different tree priors allows analyses to be tailored to a huge variety of different scenarios including:
                <ul>
                  <li class="fragment">macroevolution,</li>
                  <li class="fragment">within-population evolution,</li>
                  <li class="fragment">pathogen evolution,</li>
                  <li class="fragment">transmission tree inference,</li>
                  <li class="fragment">et cetera!</li>
                </ul>
              </li>
            </ul>

          </section>

          <section>
            <h3>Models for species trees</h3>

            <div class="figure" style="float:right;width:40%">
              <img data-src="primate_tree.jpg">
            </div>
            <div style="width:60%">
              Species trees are usually modelled using (piecewise) constant-rate birth processes such as:
              <ul>
                <li class="fragment">The Yule model (pure birth)<br> <span class="cite">[Yule, 1924]</span></li>
                <li class="fragment">The more general birth-death sampling model
                <span class="cite">[Stadler, 2010]</li>

                <li class="fragment">The fossilized birth-death model <span class="cite">[Heath, 2014]</span> directly fossil data</li>
              </ul>
            </div>

            <p class="fragment">
            The BDS tree prior is parameterized by a birth rate
            $\lambda$, death rate $\mu$ and sampling parameters $\rho$
            and $\psi$, while the Yule prior only uses $\lambda$.</p>
          </section>

          <section>
            <h3>Example Yule tree</h3>

            <br>

            <div style="text-align:center">
              <img height="450" data-src="yule_tree.svg">
            </div>
          </section>

          <section>
            <h3>Gene trees: the coalescent</h3>
            Arises as a limit of many models in population genetics.

            <div style="position:relative; width:800px; height:450px; margin:0 auto;">
              <img class="fragment current-visible" width="800" data-src="Coalescent.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment current-visible" width="800" data-src="Coalescent2.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment current-visible" width="800" data-src="Coalescent3.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
              <img class="fragment" width="800" data-src="Coalescent4.svg"
                   style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
            </div>

            <ul>
              <li class="fragment">Probability of coalescence in single generation  $\sim\binom{k}{2}\frac{1}{2N}$</li>
              <li class="fragment">Geometric distribution between coalescence events.</li>
            </ul>
          </section>
          <section>
            <h3>Example coalescent tree</h3>

            <br>

            <div style="text-align:center">
              <img data-src="coalescent_tree.svg" height="450"> <br>
            </div>
          </section>

          <section>
            <h3>Phylodynamic models</h3>

            <blockquote class="fragment">
              The term <i>phylodynamics</i> was introduced by Grenfell
              and others <span class="cite">[Science, 2004]</span> to
              describe the interaction between evolution and
              epidemiological dynamics and its effect on the phylogenies
              of rapidly evolving pathogens.
            </blockquote>

            <p class="fragment" style="margin-bottom:0px">Some models which exploit this interaction to learn about
            epidemics using genetic data are becoming common:</p>
            <ul>
              <li class="fragment">BDSKY <span class="cite">[Stadler et al., 2013]</li>
              <li class="fragment">BDSIR <span class="cite">[K&uuml;hnert et al., 2014]</li>
              <li class="fragment">Coalescent-based models <span class="cite">[Volz et al., 2009, 2012]</span></li>
              <li class="fragment">Exact models <span class="cite">[Leventhal et al., 2014]</li>
            </ul>
            <p class="fragment" style="margin-top:0px;line-spacing:1em">Allow inference of $R_0$ and other parameters, including
            prevalence curves.</p>

          </section>
        </section>

        <section>
          <section class="center">
            <h2>Bayesian Phylogenetic Inference</h2>
          </section>

          <section>
            <h3>Bringing everything together</h3>

            <p>The goal of phylogenetic inference is generally to
              learn about the phylogeny of a set of sampled genetic
              sequences.</p>

            <p class="fragment">As Bayesians we accomplish this by seeking to characterise the joint posterior:
              $$P(T,\sigma,\theta|D) = \frac{1}{P(D)} P(D|T,\sigma)P(T|\theta,l)P(\sigma)P(\theta)$$</p>

            <ul class="fragment">
              <li>$P(D|T,\sigma)$ is the <i>tree likelihood</i></li>
              <li>$P(T|\theta,l)$ is the <i>tree prior</i></li>
              <li>$P(\sigma)$ and $P(\theta)$  are the <i>parameter priors</i></li>
            </ul>

            <br><br>

            <i class="fragment">We simultaneously learn about the tree and the parameters!</i>
          </section>

          <section>
            <h3>Where is our data?</h3>

            Let's look again at the full posterior density:
            <blockquote class="fragment" style="margin-top:50px">
              $$P(T,\sigma,\theta|D) = \frac{1}{P(D)} P(D|T,\sigma)P(T|\theta,l)P(\sigma)P(\theta)$$
            </blockquote>

            <ul style="margin-top:50px">
              <li class="fragment">Where is the likelihood?</li>
              <li class="fragment">Related: where is our data?</li>
            </ul>

            <p class="fragment" style="text-align:center;color:purple;margin-top:50px">Labels such as "likelihood" and "prior" are in some sense artificial, and can become ambiguous for complex models.</p>

          </section>

          <section>
            <h3>How many trees are there?</h3>

            <div class="figure inline" style="width:550px">
              <img data-src="howmanytrees.png" style="height:550px"> 
              <div class="cite">&nbsp;</div>
            </div>

            <div class="figure inline" style="width:400px">
              <img data-src="numtrees.png" style="height:550px"> 
              <div class="cite">Felsenstein, 2004</div>
            </div>
          </section>

          <section>
            <h3>MCMC on tree space</h3>

            <p class="fragment">Our posterior is a probability distribution over trees, so
            our MCMC algorithm must be able to explore <i>tree
            space</i>.  (Much harder than exploring the real number
            line!)</p>

              <div class="figure fragment" style="text-align:center">
                <img style="width:50%" data-src="tree_moves.png">
                <div class="caption">Example tree proposals used in BEAST.</div>
            </div>
            
          </section>

          <section>
            <h3>Non-identifiability</h3>
            When the data are not informative for a particular
            parameter, the parameter is said to be non-identifiable.

            <ul>
              <li class="fragment">A classic example is the clock rate $\mu$ under a
                coalescent model when only contemporaneous sequences are
                available. In this case only the product $\mu N$ can be
                inferred.</li>

              <li class="fragment">More subtle examples are found in the structured coalescent
              and Clonal Origin models to be discussed this afternoon.</li>
            </ul>

            <p class="fragment" style="color:purple">In Bayesian analyses
            non-identifiability is not always a big problem: the worst
            that can happen is that a parameter is sampled from it's
            prior...</p>
          </section>

          <section>
            <h3>Improper Priors</h3>

            <blockquote class="alert" style="margin-top:50px">...which can be catastrophic if
            that prior is improper! In this case, the posterior will
            also be improper.</blockquote>

            <p class="fragment" style="margin-top:50px">Happily, there is a strong
            signature of an improper posterior: a variable in the MCMC
            state chain will steadily grow, eventually exceeding the
            maximum floating point value allowed by the computer.  If
            this happens, it is highly likely that the marginal
            posterior of the parameter in question is not
            normalizable.</p<

          </section>
        </section>

        <section>
          <h2>Further reading</h2>

          <div class="figure" style="width:45%;float:left;text-align:center;margin-top:50px">
            <img style="height:400px" data-src="InferringPhylogenies.jpg">
          </div> <div style="width:45%;float:right;text-align:center;margin-top:50px">
            <img style="height:400px" data-src="beast_book_cover.jpeg">
          </div>
        </section>
      </div>
      </div>
    </div>

    <script src="../reveal.js/lib/js/head.min.js"></script>
    <script src="../reveal.js/js/reveal.js"></script>

    <script>
      // More info https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
      controls: false,
      progress: false,
      history: true,
      center: false,

      transition: 'convex',

      // More info https://github.com/hakimel/reveal.js#dependencies
      dependencies: [
      { src: '../reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
      { src: '../reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
      { src: '../reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
      { src: '../reveal.js/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
      { src: '../reveal.js/plugin/zoom-js/zoom.js', async: true },
      { src: '../reveal.js/plugin/notes/notes.js', async: true },
      { src: '../reveal.js/plugin/math/math.js', async: true }
      ]
      });
    </script>
  </body>
</html>
